velero-plugins/pod/restore.go

```diff
@@ -50,8 +50,19 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	if err != nil {
 		return nil, err
 	}
-	// Check if pod has owner Refs
-	if len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" {
+
+	// get backup associated with the restore
+	backupName := input.Restore.Spec.BackupName
+	backup, err := common.GetBackup(backupName, input.Restore.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	// check for default restic flag
+	defaultVolumesToResticFlag := backup.Spec.DefaultVolumesToRestic
+
+	// Check if pod has owner Refs and defaultVolumesToRestic flag as false
+	if len(ownerRefs) > 0 && pod.Annotations[common.ResticBackupAnnotation] == "" && !*defaultVolumesToResticFlag {
```

Comment:
`defaultVolumesToResticFlag` is a `*bool`
https://github.com/vmware-tanzu/velero/blob/9b7f2da19250e8860f72cc04b6dc14db2810c425/pkg/apis/velero/v1/backup.go#L96
Should null check before dereferencing.

Reply:
You could probably just replace `!*defaultVolumesToResticFlag` with `(defaultVolumesToResticFlag == nil || !*defaultVolumesToResticFlag)`